### ISSUE TITLE
[Feature] 

### ISSUE THREAD

**Issue by kelvinwang139** [open-mmlab/mmclassification#1435]
### Branch

master branch (0.24 or other 0.x version)

### Describe the feature

support pytorch 2.0

pytorch 2.0 is official released. Does mmcv support pytorch 2.0 (officially announced by pytorch that 2.0 is 100% backward compatible by definition.)

### Will you implement it?

- [ ] I would like to implement this feature and create a PR!

**Comment by Ezra-Yu** [comment]
Yes, but please switch to dev-1.x branch. mmengine will support the pytorch 2.0
